snuba_sdk/expressions.py

```diff
@@ -155,18 +163,25 @@ def validate(self) -> None:
 
 
 @dataclass(frozen=True)
-class Function(Expression):
+class CurriedFunction(Expression):
     function: str
-    parameters: Sequence[Union[ScalarType, Column, "Function"]]
+    initializers: Optional[Sequence[Union[ScalarLiteralType, Column]]] = None
```

Comment:
Why cannot the initializer contain Function calls itself? Is it something clickhouse does not support or a constraint we are imposing. If it is the second, what's the reason?

Reply:
It is self imposed, and the reason is two fold: firstly, there is no use case for it, secondly, we don't support it in Snuba.